kubernetes/config-maps/nginx-storage/default.conf.j2

```diff
@@ -0,0 +1,96 @@
+server {
+    listen 80;
+
+    gzip                     on;
+    gzip_min_length          1000;
+    gzip_proxied             expired no-cache no-store private auth;
+    gzip_types               text/plain text/css text/javascript application/xml;
+    client_max_body_size     {{ max_upload_size }};
+
+
+    location /download/ {
+        alias /srv/storage/;
+    }
+
+    location = /upload/ {
+
+        access_by_lua_block {
+            -- Module nginx big upload support only content-type: "application/x-www-form-urlencoded"
+            check_content_type_header=ngx.req.get_headers()["Content-Type"]
+            if check_content_type_header ~= "application/x-www-form-urlencoded" then
+                ngx.status=400
+                ngx.say('Only Content-Type: "application/x-www-form-urlencoded" is allowed!')
+                ngx.exit(400)
```

Comment:
Why both `ngx.status=400` and `ngx.exit(400)`? Which one is the response status and what does the other do?

Reply:
Looks like both set the same thing but there's a big caveat: if you run `ngx.say()` nginx immediately sends response headers to the client. This means that `ngx.exit()` can no longer change the status. Nginx will print a warning if you give it a code different than the one you used in `ngx.status`:

```
[error] 12251#0: *1 attempt to set status 400 via ngx.exit after
```

See https://github.com/openresty/lua-resty-redis/issues/15#issuecomment-14747652

Reply:
One more thing. The recommented style is to use `return` with `ngx.exit()`:

``` lua
return ngx.exit(400)
```

From [docs for ngx.exit()](https://github.com/openresty/lua-nginx-module#ngxexit):
> Also note that this method call terminates the processing of the current request and that it is recommended that a coding style that combines this method call with the return statement, i.e., return ngx.exit(...) be used to reinforce the fact that the request processing is being terminated.

Reply:
Done.